slight refactor to default label

GregoryPost opened the pull request:
# Description
Slight refactor to default label initialization
Removes check testing if temp/reports is writable
Renames labeldefault.json label.json so that it will be picked up by initial DB ingestion process

# Pull Request Checklist:

# Pre-Approval

- [ ] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Hotfixes should be branched off of the `master` branch and PR'd using the **merge** option (not squashed) into the `hotfix` branch.
- [ ] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [ ] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [ ] There are no linter errors
- [ ] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [ ] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [ ] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [ ] Comment which GitHub issue(s), if any does this PR address
- [ ] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge into the `hotfix` branch, remember to use the **merge** option (i.e., no squash).
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a merge from the `hotfix` branch into the `master` branch use the **squash & merge** option
  - [ ] a subsequent PR from `master` into `Development` should be made with the **merge** option (i.e., no squash).
  - [ ] **Immediately** delete the `hotfix` branch and create a new `hotfix` branch
  - [ ] increment the Symbiota version number in the symbase.php file and commit to the `hotfix` branch.
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

GregoryPost committed:
slight refactor to default label

Atticus29 commented on content/collections/reports/.gitignore:
```diff
@@ -2,4 +2,4 @@
 !.gitignore
-!labeldefault.json
\ No newline at end of file
+!label.json
```
Given that this file *could* and is kinda meant to vary across portals and just acts to seed the row upon first use, I don't think it should be tracked by git. Thoughts? Is the labeldefault.json addition here relatively new, @egbot ?

Atticus29 replied:
Incidentally, the only mention of labeldefault.json in the entire codebase seems to be in content/collections/reports/.gitignore.... not sure whether it's cruft that needs to be removed or whether the implementation of it has been broken.

themerekat replied:
I think labeldfault.json is not supposed to be called by the code. It is a file that exists as a default/template so that someone can create a label.json based off of it.

Atticus29 replied:
Oh of course. That makes total sense. In which case, vote we keep tracking the default but not label.json.

GregoryPost replied:
It's a good question.  I was setting it to be tracked as a convenient way to distribute  it to existing portals and make sure that the value gets pulled into the DB.  Worth noodling on for a bit.

themerekat replied:
@egbot we discussed this in a meeting and agree. Mark already wrote something that converts label.json into the adminConfig table dynamicProperties field. Next step will be putting a check that if there is no label.json, it uses defaultlabel.json. I am tasked with communicating this to Mark.

Atticus29 replied:
Ok. I implemented the above in aef575194. So now, if label.json is missing and labeldefault.json is present, it will populate the dynamicProperties field of adminConfig with the contents of labeldefault.json. I also reverted the gitignore back to labeldefault.json given this change. Hope that's alright with folks!

Atticus29 replied:
I updated the update doc linked in the pull request checklist, but nothing in the readme. I'll code review asap.

Atticus29 reviewed (commented):
I want to think more about this, but I don't think label.json should be tracked.

Atticus29 committed:
revert gitignore to labeldefault.json, confirm that said json file is being tracked, and add logic to handle when only the default file is present

egbot committed:
Label generation adjustments

- Save MS Word label files to TEMP_DIR_ROOT
- Remove check to see if TEMP_DIR_ROOT directory is writable
- labeldefault.json is used to define the default label format for a new installation, or an old installation that doesn't have a custom labeljson.php file defined, which would need to be manually converted to label.json

Atticus29 commented on collections/reports/labelmanager.php:
```diff
@@ -460,27 +451,14 @@ function labelFormatChanged(selObj){
+										<div style="clear:both;padding:10px 0px">
+											<b><?php echo (isset($LANG['NOTE']) ? $LANG['NOTE'] : 'Note:') ?></b> <?php echo (isset($LANG['NOTE_1']) ? $LANG['NOTE_1'] : 'Currently, Word (DOCX) output only generates the old static label format.') ?><br/><?php echo (isset($LANG['NOTE_2']) ? $LANG['NOTE_2'] : 'Output of variable Label Formats (pulldown options) as a Word document is not yet supported.') ?><br/>
+											<?php echo (isset($LANG['NOTE_3']) ? $LANG['NOTE_3'] : 'A possible work around is to print labels as PDF and then convert to a Word doc using Adobe tools.') ?><br/>
+											<?php echo (isset($LANG['NOTE_4']) ? $LANG['NOTE_4'] : 'Another alternatively, is to output the data as CSV and then setup a Mail Merge Word document.') ?>
```
Typo here. "Another alternatively," should change to "Another alternative", and the comma should be removed.

Atticus29 reviewed (changes requested):
Code looks good to me other than typo in the translation file.

egbot committed:
Fix grammar error

Atticus29 reviewed (approved):
LGTM!

egbot committed:
Misc adjustments

- Since value is going in the database, no need to save as pretty_print, which only takes up space. That was only being done so that json could be easily read within the text file.
- Remove multiple spaces from json input file
